### PR TITLE
Add support for `ensure_future` for RUF006

### LINKED ISSUE #2941: Add `asyncio.ensure_future` support to `RUF006`
[asyncio.ensure_future](https://docs.python.org/3/library/asyncio-future.html#asyncio.ensure_future) can be garbage collected similar to `create_task`, causing the task to never finish.

From asyncio docs:
> Save a reference to the result of this function, to avoid a task disappearing mid-execution.

### PR DESCRIPTION
**Pull request by Lunarmagpie**
apologies if I did the snapshots wrong lol. This is my first time using the tool.

closes #2941

### CODE REVIEW

File: crates/ruff/src/rules/ruff/rules/asyncio_dangling_task.rs [hunk 1]
```diff
@@ -55,7 +56,7 @@ define_violation!(
 impl Violation for AsyncioDanglingTask {
     #[derive_message_formats]
     fn message(&self) -> String {
-        format!("Store a reference to the return value of `asyncio.create_task`")
+        format!("Store a reference to the return value of `asyncio.create_task` and `asyncio.ensure_future`")
```
Comment (charliermarsh) [inline crates/ruff/src/rules/ruff/rules/asyncio_dangling_task.rs:59]: Let's tailor this message based on the function call. If you take a look at the `DeferralKeyword` enum in `crates/ruff/src/rules/pyflakes/rules/yield_outside_function.rs`, we can do the same thing here: add it as a member on the rule struct, then include it in the format message.

Comment (Lunarmagpie) [reply crates/ruff/src/rules/ruff/rules/asyncio_dangling_task.rs]: Change is made! :smile:

**Comment by Lunarmagpie** [comment]
Is there anything special I have to do to get both keywords to appear properly in the README?

**Comment by charliermarsh** [comment]
Na, the README only shows the message _format_, not the realized version with the placeholders filled in. So this is by design, even if somewhat suboptimal for certain rules.

File: crates/ruff/src/rules/ruff/rules/asyncio_dangling_task.rs [hunk 2]
```diff
@@ -49,13 +51,31 @@ define_violation!(
...
     #[derive_message_formats]
     fn message(&self) -> String {
-        format!("Store a reference to the return value of `asyncio.create_task`")
+        let AsyncioDanglingTask { method } = self;
+        format!("Store a reference to the return value of `asyncio.{method}`")
+    }
+}
+
+#[derive(Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub enum Method {
```
Comment (charliermarsh) [inline crates/ruff/src/rules/ruff/rules/asyncio_dangling_task.rs:68]: I just renamed this to something more appropriate for the specific check.

**Pull request merged**